cockatrice/src/messagelogwidget.cpp

```diff
@@ -3,190 +3,87 @@
-{
-    appendMessage(message, 0, spectatorName, spectatorUserLevel, userPrivLevel, false);
-}
-
-void MessageLogWidget::logShuffle(Player *player, CardZone *zone)
-{
-    soundEngine->playSound("shuffle");
-    if (currentContext != MessageContext_Mulligan) {
-        appendHtmlServerMessage(
-         tr("%1 shuffles %2.")
-         .arg(sanitizeHtml(player->getName()))
-         .arg(zone->getTranslatedName(true, CaseShuffleZone)));
-    }
+    static const QString constant("stack");
+    return constant;
 }
 
-void MessageLogWidget::logRollDie(Player *player, int sides, int roll)
-{
-    QString coinOptions[2] = {tr("Heads")+" (1)", tr("Tails")+" (2)"};
-    soundEngine->playSound("roll_dice");
-
-    if (sides == 2)
-        appendHtmlServerMessage(tr("%1 flipped a coin. It landed as %2.").arg(sanitizeHtml(player->getName())).arg("<font color=\"blue\">" + coinOptions[roll - 1] + "</font>"));
-    else
-        appendHtmlServerMessage(tr("%1 rolls a %2 with a %3-sided die.").arg(sanitizeHtml(player->getName())).arg("<font color=\"blue\">" + QString::number(roll) + "</font>").arg("<font color=\"blue\">" + QString::number(sides) + "</font>"));
-}
-
-void MessageLogWidget::logDrawCards(Player *player, int number)
+QString MessageLogWidget::sanitizeHtml(QString dirty) const
 {
-    if (currentContext == MessageContext_Mulligan)
-        mulliganPlayer = player;
-    else {
-        soundEngine->playSound("draw_card");
-        appendHtmlServerMessage(tr("%1 draws %2 card(s).").arg(sanitizeHtml(player->getName())).arg("<font color=\"blue\">" + QString::number(number) + "</font>"));
-    }
+    return dirty
+            .replace("&", "&amp;")
+            .replace("<", "&lt;")
+            .replace(">", "&gt;")
+            .replace("\"", "&quot;");
 }
 
-void MessageLogWidget::logUndoDraw(Player *player, QString cardName)
+QString MessageLogWidget::cardLink(const QString cardName) const
 {
-    if (cardName.isEmpty())
-        appendHtmlServerMessage(tr("%1 undoes their last draw.").arg(sanitizeHtml(player->getName())));
-    else
-        appendHtmlServerMessage(tr("%1 undoes their last draw (%2).").arg(sanitizeHtml(player->getName())).arg(QString("<a href=\"card://%1\">%2</a>").arg(sanitizeHtml(cardName)).arg(sanitizeHtml(cardName))));
+    return QString("<i><a href=\"card://%1\">%2</a></i>")
+            .arg(cardName)
+            .arg(cardName);
 }
 
-QPair<QString, QString> MessageLogWidget::getFromStr(CardZone *zone, QString cardName, int position, bool ownerChange) const
+QPair<QString, QString> MessageLogWidget::getFromStr(
+        CardZone *zone,
+        QString cardName,
+        int position,
+        bool ownerChange) const
 {
     bool cardNameContainsStartZone = false;
     QString fromStr;
-    QString startName = zone->getName();
+    QString zoneName = zone->getName();
 
-    if (startName == "table")
+    if (zoneName == tableConstant())
```

Comment:
This is the construct on first use idiom.
http://stackoverflow.com/questions/335369/finding-c-static-initialization-order-problems

I dont feel adding "global" consts here is a good idea as you suggest. Instead it might be better to start a `constants.h/cpp` that can be accessed from anywhere.